src/component/page-list/page-tile-container.tsx

```diff
@@ -0,0 +1,94 @@
+import * as MobX from 'mobx';
+import { observer } from 'mobx-react';
+import { PageRef } from '../../store/page/page-ref';
+import { PageTileComposite } from './page-tile-composite';
+import * as React from 'react';
+import { Store } from '../../store/store';
+
+export interface PageTileContainerProps {
+	focused: boolean;
+	onClick: React.MouseEventHandler<HTMLElement>;
+	page: PageRef;
+}
+
+@observer
+export class PageTileContainer extends React.Component<PageTileContainerProps> {
+	@MobX.observable public editable: boolean = false;
+	@MobX.observable public inputValue: string = '';
+
+	public constructor(props: PageTileContainerProps) {
+		super(props);
+		this.inputValue = this.inputValue ? this.inputValue : this.props.page.getName();
+
+		this.handleChange = this.handleChange.bind(this);
+		this.handleClick = this.handleClick.bind(this);
+		this.handleDoubleClick = this.handleDoubleClick.bind(this);
+		this.handleKeyDown = this.handleKeyDown.bind(this);
+	}
+
+	@MobX.action
+	protected handleBlur(): void {
+		this.editable = false;
```

Comment:
Blur should commit the change, shouldn't it?

Reply:
When the input element is no longer `:focus` it should just set the `this.editable` flag to false. Changes are not saved.

Is that what you mean @TheReincarnator?

Reply:
Yes but at least on Windows, the common patterns is that blurring a rename field commits it, not aborts it. How does the Finder behave?

Reply:
In the finder the changes are saved on blur